src/Service/ServiceManagerConfig.php

```diff
@@ -150,6 +150,7 @@ public function __construct(array $config = [])
                 // implementations; they're dealt with later.
                 if ($instance instanceof ServiceLocatorAwareInterface
                     && ! $instance instanceof AbstractPluginManager
+                    && ! $instance->getServiceLocator()
```

Comment:
Is this the original behavior? Can't remember if we did overwrite the `ServiceLocatorInterface` instance each time or not (see previous framework versions). We can't change this, if we did overwrite it at each occasion.